assets/javascripts/discourse/components/events-calendar-subscription.js.es6

```diff
@@ -4,23 +4,47 @@ import computed from 'ember-addons/ember-computed-decorators';
 export default DropdownSelectBoxComponent.extend({
   classNames: ["events-calendar-subscription"],
   rowComponent: "events-calendar-subscription-row",
+  filterComponent: "events-calendar-subscription-warning",
 
-  @computed
-  content() {
+  @computed('authSuffix')
+  content(authSuffix) {
     const baseUrl = window.location.host + window.location.pathname;
     const timeZone = moment.tz.guess();
     return [
       {
-        id: `webcal://${baseUrl}.ics?time_zone=${timeZone}`,
+        id: `webcal://${baseUrl}.ics?time_zone=${timeZone}${authSuffix}`,
         name: I18n.t('events_calendar.ical')
       },
       {
-        id: `${baseUrl}.rss?time_zone=${timeZone}`,
+        id: `${baseUrl}.rss?time_zone=${timeZone}${authSuffix}`,
         name: I18n.t('events_calendar.rss')
       }
     ];
   },
 
+  @computed('userApiKey', 'category', 'siteSettings.login_required')
+  authSuffix(userApiKey, category, loginRequired) {
+    // only private sites need login
+    if (!loginRequired) return "";
```

Comment:
@haveyaseen @fzngagan This logic doesn't handle the case of when login is not required, but the calendar is in a private category. We can have a single if statement here:
```
if (!loginRequired && (!category || !category.read_restricted)) return "";
```

Reply:
You're saying links to private categories' feeds need to have an auth suffix even when the Discourse instance is public, right? If that's the case I now implemented that. Also had to change it in the parent component that makes the AJAX request to fetch the API key(s).